extension.js

```diff
@@ -168,6 +179,41 @@ const ExtensionLayout = new Lang.Class({
     GLib.spawn_command_line_async(cmd);
   },
 
+  _findNewStreamerEntries:function(lastList, currentList, detectGameChange) {
+    detectGameChange = detectGameChange || false;
+    if ( lastList.length == 0 )
+      return currentList;
+
+    let streamers = new Map();
+    lastList.forEach( ({streamer, game}) => streamers.set(streamer, game) );
+
+    return currentList.filter( ({streamer, game}) => {
+        if (!streamers.has(streamer))
+            return true;
+
+        if (detectGameChange &&
+            game != streamers.get(streamer))
+            return true;
```

Comment:
isn't this just `return (detectGameChange && game != streamers.get(streamer))`?

or even `return (!streamers.has(streamer) || (detectGameChange && game != streamers.get(streamer)))`

I'm not always sure which one is more readable. For me the if-constructs always implies that something is actually happening and the decision has impact on this here function and not the outside, if you understand my meaning. But if you don't think so, I'm fine with either.

Reply:
> or even `return (!streamers.has(streamer) || (detectGameChange && game != streamers.get(streamer)))`
> I'm not always sure which one is more readable.

I did combine all of the conditions when I first wrote this, but I found it more readable to separate out the conditions by flags
```js
if (!streamer.has(streamer) || (!flag || (flag && conditions)) || (!flag2 && (flag2 && (condition2 == blag.name || condition2.alternate == blag.altName))
```
```js
if (!streamer.has(streamer))
...

if (flag &&
    conditions)
...

if (flag2 &&
    (  condition2 == blag.name
    || condition2.alternate == blag.altName
    ))
...
```

I also found it more readable to do `if (flag &&\n...` because I felt that this style makes it easier to skim the checks when the `if (flag...` is its own line because the conditions are not guaranteed to be super short. I could just put the conditions in a nested `if`, if you want.

Let me know if any of that seems funny to you.

Reply:
Nested would be the worst of the possibilities.

If you're fine with the return true-pattern, then I guess, I am as well.

Reply:
Ohh, sorry, I didn't read this question correctly

> isn't this just return (detectGameChange && game != streamers.get(streamer))?

No... Yes. It is effectively *that*, right now.

I wanted to make it easy to add more things to check for. If you do return the value of `(flag && condition)` then it unconditionally returns at that point, so you can't add other things to check for (unless oneline, which this is avoiding).

```js
fn {
    // doesn't match, allows continued matching
    if (false_condition)
        return true;

    // returns unconditionally
    return (flag && conditions);

    // unreachable
    return (flag2 && conditions2);
    ...
```